src/osd/ECBackend.cc

```diff
@@ -2342,13 +2342,23 @@ int ECBackend::send_all_remaining_reads(
   GenContext<pair<RecoveryMessages *, read_result_t& > &> *c =
     rop.to_read.find(hoid)->second.cb;
 
+  // (Note cuixf) If we need to read attrs and we read failed, try to read again.
+  bool want_attrs = 
+    rop.to_read.find(hoid)->second.want_attrs && 
+    (!(rop.complete[hoid].attrs) ||
+    ((rop.complete[hoid].attrs) && !(*(rop.complete[hoid].attrs)).size()));
```

Comment:
nit: could simplify ```(rop.complete[hoid].attrs) && !(*(rop.complete[hoid].attrs)).size())``` to ```rop.complete[hoid].attrs.empty()```

Reply:
simplify (rop.complete[hoid].attrs) && !(*(rop.complete[hoid].attrs)).size()) to rop.complete[hoid].attrs.empty() may lead to something error. if the rop.complete[hoid].attrs doesn't have pointed to a map<string, bufferlist>, it may crash.

Reply:
replacing the commit would be easiest, thanks!